phpBB/phpbb/captcha/plugins/qa.php

```diff
@@ -100,6 +100,28 @@ function init($type)
 			$db->sql_freeresult($result);
 		}
 
+		// final fallback to any language
+		if (!sizeof($this->question_ids))
+		{
+			$this->question_lang = '';
+
+			$sql = 'SELECT q.question_id, q.lang_iso
+				FROM ' . $this->table_captcha_questions . ' q, ' . $this->table_captcha_answers . ' a
+				WHERE q.question_id = a.question_id
+				GROUP BY lang_iso';
+			$result = $db->sql_query($sql, 7200);
+
+			while ($row = $db->sql_fetchrow($result))
+			{
+				if (empty($this->question_lang))
+				{
+					$this->question_lang = $row['lang_iso'];
+				}
+				$this->question_ids[$row['question_id']] = $row['question_id'];
+			}
+			$db->sql_freeresult($result);
```

Comment:
We only do this if:
- the user selected language does not have questions available
- the default language does not have questions available

In order to be able to properly select questions and fitting answers, we need to have a valid language selected (no array, just a lang iso). We could of course also select a language with picking a random question using array_lang() and then select its lang iso.